### PR TITLE
Bookmarking feature

### PR DESCRIPTION
**Pull request by joutad**

### CODE REVIEW

File: core/urls.py [hunk 1]
```diff
@@ -5,13 +5,22 @@
...
     path("hackathons/", views.HackathonsPage.as_view(), name="hackathons"),
     path("add-hackathons/", views.addHackathons, name="add_hackathons"),
     path("calendar/", views.calendar, name="calendar"),
-    path("save/<int:hackathon_id>/", views.save_hackathon, name="save_hackathon"),
     path(
         "saved_hackathons/",
         views.SavedHackathonsPage.as_view(),
         name="saved_hackathons",
     ),
+    path(
```
Comment (JasonLovesDoggo) [inline core/urls.py:16]: Nitpick but place related views together

File: core/views.py [hunk 2]
```diff
@@ -73,6 +74,17 @@ def save_hackathon(request, hackathon_id):
...
 
 
+@login_required
+def unsave_hackathon(request: HttpRequest, hackathon_id):
+    if request.method == "POST":
+        user = request.user
+        hackathon = get_object_or_404(Hackathon, id=hackathon_id)
+        user.saved.remove(hackathon)
+
+    print(request.path)
```
Comment (JasonLovesDoggo) [inline core/views.py]: remove debug print

File: core/views.py [hunk 3]
```diff
@@ -73,6 +74,17 @@ def save_hackathon(request, hackathon_id):
...
+    return redirect("saved_hackathons")
```
Comment (JasonLovesDoggo) [inline core/views.py]: we dont want to redirect or anything, this is a background call.

File: core/views.py [hunk 4]
```diff
@@ -73,6 +74,17 @@ def save_hackathon(request, hackathon_id):
     return redirect("hackathons")
 
 
+@login_required
+def unsave_hackathon(request: HttpRequest, hackathon_id):
+    if request.method == "POST":
+        user = request.user
+        hackathon = get_object_or_404(Hackathon, id=hackathon_id)
+        user.saved.remove(hackathon)
```
Comment (JasonLovesDoggo) [inline core/views.py]: This doesn't handle the case that this endpoint gets called twice (due to wacky internet) and returns a 500 as the hackathon is already unsaved.

File: core/views.py [hunk 5]
```diff
@@ -73,6 +74,17 @@ def save_hackathon(request, hackathon_id):
     return redirect("hackathons")
```
Comment (JasonLovesDoggo) [inline core/views.py]: please apply the same changes to the save hackathon view

**Review by JasonLovesDoggo — DISMISSED** [review]
Really just the two comments about the unsave_hackathon view (which should also be applied to save_hackathon)

File: core/views.py (hunk 4 above)
Comment (joutad) [reply core/views.py]: Not sure how to handle this one

Comment (JasonLovesDoggo) [reply core/views.py]: You can simply add a if statement 

Attached is an example 
```py
from myapp.models import Article, Publication

article = Article.objects.get(id=1)
publication = Publication.objects.get(id=2)

if publication in article.publications.all():
    article.publications.remove(publication)
```

Comment (joutad) [reply core/views.py]: I've only tried something like this with my superuser and it seems to not work... I don't remember the exact message but I'll keep you posted

Comment (JasonLovesDoggo) [reply core/views.py]: I don't get what you mean about your superuser but yeah, keep us posted!

Comment (joutad) [reply core/views.py]: Nevermind, everything worked okay, I was initially getting the list incorrectly, that's all

**Pull request merged**